api/client/version.go

```diff
@@ -12,34 +12,71 @@ import (
 	"github.com/docker/docker/utils"
 )
 
+var VersionTemplate = `Client:
+ Version:      {{.Client.Version}}
+ API version:  {{.Client.ApiVersion}}
+ Go version:   {{.Client.GoVersion}}
+ Git commit:   {{.Client.GitCommit}}
+ Built:        {{.Client.BuildTime}}
+ OS/Arch:      {{.Client.Os}}/{{.Client.Arch}}{{if .Client.Experimental}}
+ Experimental: {{.Client.Experimental}}{{end}}{{if .ServerOK}}
+
+Server:
+ Version:      {{.Server.Version}}
+ API version:  {{.Server.ApiVersion}}
+ Go version:   {{.Server.GoVersion}}
+ Git commit:   {{.Server.GitCommit}}
+ Built:        {{.Server.BuildTime}}
+ OS/Arch:      {{.Server.Os}}/{{.Server.Arch}}{{if .Server.Experimental}}
+ Experimental: {{.Server.Experimental}}{{end}}{{end}}`
+
+type VersionData struct {
+	Client   types.Version
+	ServerOK bool
+	Server   types.Version
+}
+
 // CmdVersion shows Docker version information.
 //
 // Available version information is shown for: client Docker version, client API version, client Go version, client Git commit, client OS/Arch, server Docker version, server API version, server Go version, server Git commit, and server OS/Arch.
 //
 // Usage: docker version
-func (cli *DockerCli) CmdVersion(args ...string) error {
+func (cli *DockerCli) CmdVersion(args ...string) (err error) {
 	cmd := cli.Subcmd("version", nil, "Show the Docker version information.", true)
+	tmplStr := cmd.String([]string{"f", "#format", "-format"}, "", "Format the output using the given go template")
```

Comment:
@phemmer i just noticed: you shouldn't use `#format`, that's for old flags, you are introducing a new one here.